driver/bpf/fillers.h

```diff
@@ -2316,7 +2316,7 @@ FILLER(proc_startupdate, true)
 	 * ptid
 	 */
 	real_parent = _READ(task->real_parent);
-	pid_t ptid = _READ(real_parent->pid);
```

Comment:
@Andreagit97 This effectively means we are dropping thread relationship awareness. Have we considered the consequences of this in the libs code/thread table esp wrt various state-changing events like proc fork, exec, exit etc. Can you please share your thoughts about this, thanks!

Reply:
yep sure! The events involved in this change would be `execve/execveat` and `clone/clone3/vfork/fork`, so the parser involved are `parse_clone_exit` and `parse_execve_exit`. In sinsp changes are quite simple we have just to fill the `ptid` field of the thread info with this new value (`ptid` is no more a good name, probably `ppid` would be better!) After the initial `/proc` scan we populate new processes with `parse_clone_exit`, if this parser is able to put the correct information in place we have done! This change will bring to main advantages:
* **Consistency** between `/proc` scan and what we receive from our drivers. Without this patch, we cannot be sure about the content of `ptid` in userspace, if the thread info is populated by `/proc` we will have the parent pid while if it is populated by our drivers we will have the parent tid.
* **Easy management and clarity**. In UNIX a not-leader thread cannot be the parent of other threads, not sure why we introduced this concept in our libraries but this requires us a bigger effort to keep in sync with the actual thread status in the kernel. For example, a thread could die while the thread leader is still alive, so our child created by the thread could remain an orphan even if its parent process is still there

Reply:
Found a great blog post https://mozillazg.com/2022/05/ebpf-libbpfgo-get-process-info-en.html.
In addition, these changes ensure we adopt the same approach when capturing the ppid as seen from the host namespace vs pid namespace (if applicable in containers etc). e.g. explore `extract__xid_nr_seen_by_namespace` in modern_bpf.

Reply:
@Andreagit97 perhaps good to add some more comments throughout and why not go ahead and rename `ptid` to `ppid` right away?

Reply:
as discussed here this PR will include just tests :)